source/env_tutorial/minigrid_zh.rst

```diff
@@ -0,0 +1,307 @@
+MiniGrid
+~~~~~~~
+
+概述
+=======
+
+MiniGrid, 即最小化的格子世界环境，是经典的稀疏奖励的离散动作空间强化学习环境，常作为离散动作空间条件下稀疏强化学习算法的基准测试环境。
+在该游戏中，智能体需要学习在一个离散动作集合中选择合适的动作，在方格世界中完成移动，获取钥匙，开门等一系列序列决策以到达目标的位置。
+它有许多不同的实现版本，这里主要介绍
+\ `MiniGrid <https://github.com/maximecb/gym-minigrid>`__, 因为其实现简单、轻量、代码依赖少、容易安装。
+它包括MiniGrid-Empty-8x8-v0, MiniGrid-FourRooms-v0, MiniGrid-DoorKey-16x16-v0, MiniGrid-KeyCorridorS3R3-v0,
+MiniGrid-ObstructedMaze-2Dlh-v0, MiniGrid-ObstructedMaze-Full-v0等一系列环境, 下图所示为其中的MiniGrid-DoorKey-16x16-v0游戏。
+
+.. image:: ./images/MiniGrid-DoorKey-16x16-v0.png
+   :align: center
+
+安装
+====
+
+安装方法
+--------
+
+用户可以选择通过pip一键安装或git clone代码库后pip本地安装.
+
+注：如果用户没有root权限，请在install的命令后面加上--user
+
+.. code:: shell
+
+   # Method1: Install Directly
+   pip install gym-minigrid
+   # Method2: First clone this repository and install the dependencies with pip
+   git clone https://github.com/maximecb/gym-minigrid.git
+   cd gym-minigrid
+   pip install -e .
+
+验证安装
+--------
+
+安装完成后，可以在Python命令行中运行如下命令, 如果显示出游戏的交互界面，则证明安装成功：
+
+.. code:: python
+    cd gym-minigrid
+   ./manual_control.py --env MiniGrid-Empty-8x8-v0
+
+.. _变换前的空间原始环境）:
+
+变换前的空间（原始环境）
+========================
+
+.. _观察空间-1:
+
+观察空间
+--------
+
+- 以MiniGrid-Empty-8x8-v0为例，
+
+.. code:: python
+   env = gym.make('MiniGrid-Empty-8x8-v0')
+   obs1 = env.reset()  # obs: {'image': numpy.ndarray (7, 7, 3),'direction': ,'mission': ,}
+   env = RGBImgPartialObsWrapper(env) # Get pixel observations
+   obs2 = env.reset()  # obs: {'mission': ,'image': numpy.ndarray (56, 56, 3)}
+   env = ImgObsWrapper(env) # Get rid of the 'mission' field
+   obs3 = env.reset()  # obs: numpy.ndarray (56, 56, 3)
+
+   # 不能在使用上述Wrapper后再使用此FlatObsWrapper，应该单独使用
+   env = gym.make('MiniGrid-Empty-8x8-v0')
+   env = FlatObsWrapper(env)
+   obs4 = env.reset()  # obs: numpy.ndarray  (56, 56, 3)
+
+
+- obs1为一个\ ``dict``，包含\ ``image``, \ ``direction``, \ ``mission``,这3个字段, 其中\ ``image``\字段是一个shape为(7, 7, 3)的\ ``numpy.ndarray``，数据类型为\ ``uint8``
+  (7，7)表示只观测到附近7x7方格内的世界(因为环境是部分可观测的)，3表示每个小格对应一个3维的描述向量，注意这里不是真正的图像；\ ``direction``\字段是给出一个指导性的方向；
+  \ ``mission``\字段是一个文本字符串，描述了智能体应该达到什么样的目标以获得奖励.
+- 用户如果想使用真正的像素形式的图像，需要通过\ ``RGBImgPartialObsWrapper``\对env进行封装，obs2为一个\ ``dict``，包含\ ``mission``, \ ``image``\这2个字段，其中\ ``image``\字段是一个shape为(56, 56, 3)的\ ``numpy.ndarray``\，数据类型为\ ``uint8``
+  是环境是部分可观测的真正的图像；
+- 再通过\ ``ImgObsWrapper``\后，obs3是一个\ ``numpy.ndarray``，shape为(56, 56, 3)，数据类型为\ ``uint8``
+- 我们的代码库使用第4种\ ``FlatObsWrapper`` \方式，这种方式将\ ``mission``\字段中的任务字符串用one-hot的方式编码，
+  并将其与\ ``image``\字段内容拼接成一个\ ``numpy.ndarray``\obs4，其shape为(2739，)，数据类型为\ ``float32``
```

Comment:
Shall we put obs all the way through instead of denoting each obs wrapped by a specific wrapper  by obs1, obs2, obs3 etc.